src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -216,7 +218,31 @@ func (a *StorageMinerActorCode_I) _submitPowerReport(rt Runtime, lastPoStRespons
 			Params_: processDealExpirationParam,
 		})
 	}
+}
+
+func (a *StorageMinerActorCode_I) _claimDealPayments(rt Runtime) {
```

Comment:
This is a larger issue to be zoomed out but methods like these are going to be very expensive to run if run too often. (obviously has been brought up before)

Reply:
To that larger design question:

This method is called on `_onSuccessfulPoSt`. My q is this. Why do we need to cycle through every deal in the network on every successful post? Why not just deal with the deals for the sector that has been posted only?

Or conversely only run this version of `_claimDealPayments`every X epochs (via cron) rather than onSuccessfulPoSt.

I think we should move toward one of those two solutions.

Reply:
I dont think we are cycling through every deal in the network on every successful PoSt. This is every active deal in this StorageMinerActor that just submitted its PoSt successfully.